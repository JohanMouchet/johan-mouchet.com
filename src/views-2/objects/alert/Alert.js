// @flow

import * as React from "react";
import cx from "classnames";
import "./Alert.scss";

type Props = {
  type?: "info" | "success" | "warning" | "danger",
  content?: Node,
  hidden: boolean,
  closable?: boolean,
  metadata?: {
    contextClass?: string,
    additionalClasses?: Array<string>
  }
};

const Alert = (props: Props) => {
  const { type, content, hidden, closable, metadata = {} } = props;

  const AlertClass = cx(
    "o-alert",
    `o-alert--${type}`,
    {
      "is--visible": !hidden,
      "js-alert": closable
    },
    "wow",
    metadata.contextClass,
    metadata.additionalClasses
  );

  const getIcon = type => {
    if (type === "info") {
      return "info";
    } else if (type === "success") {
      return "check_circle";
    } else if (type === "warning") {
      return "warning";
    } else if (type === "danger") {
      return "error";
    }
  };

  return (
    <div className={AlertClass}>
      <i className="o-alert__icon material-icons">{getIcon(type)}</i>
      <div className="o-alert__content">{content}</div>
      {closable && (
        <button className="o-alert__close js-alert-close">&#x2716;</button>
      )}
    </div>
  );
};

Alert.defaultProps = {
  type: "info"
};

export default Alert;
