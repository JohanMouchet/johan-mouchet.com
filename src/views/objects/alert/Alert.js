// @flow

import * as React from "react";
import cx from "classnames";
import "./___.scss";

/**
 * Alert
 *
 * @param {string}  [type = info]                - Alert type
 * @param {string}  content                      - Alert content
 * @param {boolean} [hidden = false]             - Alert visibility
 * @param {boolean} [closable]                   - Alert closable capacity
 * @param {string}  [metadata.contextClass]      - Alert context class
 * @param {array}   [metadata.additionalClasses] - Alert additional classes
 */

const ___ = (props: Props) => {
  const {
    prop1,
    prop2
  } = props;

  const ___Class = cx(
	"o-alert",
	props.metadata.contextClass,
	props.metadata.additionalClasses,
    {
      prop1: "___--prop1"
    }
  );

  return (
	{% if props.type == 'info' %}
		{% set icon = 'info' %}
	{% elseif props.type == 'success' %}
		{% set icon = 'check_circle' %}
	{% elseif props.type == 'warning' %}
		{% set icon = 'warning' %}
	{% elseif props.type == 'danger' %}
		{% set icon = 'error' %}
	{% endif %}

	<div className={`${___Class} o-alert--${ props.type|default('info') } ${ not props.hidden && 'is--visible'} ${ props.closable && 'js-alert'} wow`}>
		<i className={`${'o-alert__icon'|contextClass(contextClass) } material-icons`}>{ icon|default('info') }</i>
		<div className={'o-alert__content'|contextClass(contextClass)}>{ props.content|raw }</div>
		{% if props.closable %}
			<button className={`${'o-alert__close'|contextClass(contextClass) } js-alert-close`}>&#x2716;</button>
		{% endif %}
	</div>
  );
};

___.defaultProps = {
  prop1: 2
};

export default ___;
