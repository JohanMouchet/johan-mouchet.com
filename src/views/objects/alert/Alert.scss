@import "assets/styles/settings/_settings";
@import "assets/styles/tools/_tools";

.o-alert {
  align-items: flex-start;
  margin: t-vertical-rhythm(1) 0;
  padding: t-vertical-rhythm(0.75) ($grid-gutter / 2);
  border-left: 2px solid;
  background-color: $s-color-gray-lighter;
  animation: 0.3s ease-out both;
  animation-name: u-animation--fade-in, alert-animation;

  /* Animation */
  @keyframes alert-animation {
    from {
      transform: translateY(t-vertical-rhythm(1));
    }
  }

  /* Icon */
  &__icon {
    flex-shrink: 0;

    svg {
      font-size: t-modular-scale(1);
      vertical-align: middle;
    }
  }

  /* Content */
  &__content {
    padding-left: 0;

    *:first-child {
      margin-top: 0;
    }

    *:last-child {
      margin-bottom: 0;
    }
  }

  /* Close */
  &__close {
    flex-shrink: 0;

    &-button {
      background-color: transparent;
      border: 0;
      padding: 0;
      outline: 0;
      all: unset;

      @include t-link($s-color-text-dimmed, "none", $hover-decoration: "none");

      cursor: pointer;

      svg {
        font-size: t-modular-scale(1);
        vertical-align: middle;
      }
    }
  }

  /* Variants */
  &--success {
    border-color: $s-color-success;
  }

  &--success & {
    &__icon {
      color: $s-color-success;
    }
  }

  &--info {
    border-color: $s-color-info;
  }

  &--info & {
    &__icon {
      color: $s-color-info;
    }
  }

  &--warning {
    border-color: $s-color-warning;
  }

  &--warning & {
    &__icon {
      color: $s-color-warning;
    }
  }

  &--danger {
    border-color: $s-color-danger;
  }

  &--danger & {
    &__icon {
      color: $s-color-danger;
    }
  }

  &--closable & {
    &__content {
      padding-right: 0;
    }
  }

  &--fixed-bottom,
  &--fixed-bottom-right {
    position: fixed;
    bottom: 0;
    right: 0;
    margin-bottom: 0;
    box-shadow: map-get($s-shadows, md);
    z-index: map-get($s-layers, "alert");
  }

  &--fixed-bottom-right {
    @media (min-width: map-get($s-breakpoints, sm)) {
      max-width: map-get($s-breakpoints, sm);
      margin-right: $grid-gutter;
    }
  }
}
