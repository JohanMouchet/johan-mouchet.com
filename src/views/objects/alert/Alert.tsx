import React from "react";
import { useState } from "react";
import cx from "classnames";
import {
  IconCheckCircle,
  IconExclamationCircle,
  IconExclamationTriangle,
  IconInfoCircle,
  IconX,
} from "views/objects";
import styles from "./Alert.module.scss";

type Props = {
  isOpen?: boolean;
  closable?: boolean;
  variant?: "info" | "success" | "warning" | "danger";
  position?: "fixed-bottom" | "fixed-bottom-right";
  className?: string | string[] | { [key: string]: boolean };
  children: React.ReactNode;
};

const Alert: React.FC<Props> = ({
  isOpen = true,
  closable,
  variant = "info",
  position,
  className,
  children,
}) => {
  const [open, setOpenState] = useState(isOpen);

  const classNames = cx(
    styles["alert"],
    closable && styles["alert--closable"],
    variant && styles[`alert--${variant}`],
    position && styles[`alert--${position}`],
    "container",
    "u-animation",
    "u-animation--fade-in-up",
    className
  );

  const getIcon = (variant: string) => {
    if (variant === "info") {
      return <IconInfoCircle />;
    } else if (variant === "success") {
      return <IconCheckCircle />;
    } else if (variant === "warning") {
      return <IconExclamationTriangle />;
    } else if (variant === "danger") {
      return <IconExclamationCircle />;
    }
  };

  return !open ? null : (
    <div className={classNames}>
      <div className="grid grid--noWrap-@xs">
        <div className={`${styles.icon} cell`} aria-label={variant}>
          {getIcon(variant)}
        </div>
        <div className={`${styles.content} cell cell--@xs`}>{children}</div>
        {closable && (
          <div className={`${styles.close} cell`}>
            <button
              className={styles["close-button"]}
              aria-label="Close"
              onClick={() => setOpenState(!open)}
              type="button"
            >
              <IconX />
            </button>
          </div>
        )}
      </div>
    </div>
  );
};

export default Alert;
