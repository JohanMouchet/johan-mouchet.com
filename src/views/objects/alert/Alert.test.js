import React from "react";
import { shallow } from "enzyme";
import Alert from ".";

it("Renders <Alert /> without crashing", () => {
  shallow(<Alert />);
});
