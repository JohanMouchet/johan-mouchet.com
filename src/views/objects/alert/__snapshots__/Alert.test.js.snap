// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Alert /> without crashing 1`] = `
<div
  className="o-alert o-alert--info container wow"
>
  <div
    className="grid grid--noWrap-@xs"
  >
    <div
      aria-label="info"
      className="o-alert__icon cell"
    >
      <ForwardRef(SvgInfoCircle) />
    </div>
    <div
      className="o-alert__content cell cell--@xs"
    />
  </div>
</div>
`;
